pkg/reconciler/revision/resources/deploy.go

```diff
@@ -230,6 +230,12 @@ func makeContainer(container corev1.Container, rev *v1.Revision) corev1.Containe
 	container.Lifecycle = userLifecycle
 	container.Env = append(container.Env, getKnativeEnvVar(rev)...)
 
+	// Set PSP requirements explicitly to avoid failures in case `pod-security.kubernetes.io/enforce=restricted` is used
+	// at the user workload namespace
+	container.SecurityContext.AllowPrivilegeEscalation = ptr.Bool(false)
```

Comment:
We don't allow setting this anyway. So we enforce the default here.

Reply:
@mattmoor just made `allowPrivilegeEscalation` pass-through in #13395 , though we could force it to `false` as we do here instead.

Reply:
I sent #13401 to allow setting this in the Revision Spec (e.g. in Service).